src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -23,6 +24,17 @@ public class ApiHubConfiguration : IExtensionConfigProvider, IFileTriggerStrateg
 
         ApiHubLogger _logger;
 
+        private ConnectionFactory ConnectionFactory { get; }
+
+        /// <summary>
+        /// Creates a new instance of this class.
+        /// </summary>
+        /// <param name="connectionFactory">The factory used to create connections</param>
+        public ApiHubConfiguration(ConnectionFactory connectionFactory = null)
```

Comment:
It seemed that both the extensions and script engine lacks testability features. A bunch of objects (services) are new-ed inside without the ability to replace with mocks/fakes etc. This was the minimum I could do to improve that and it is great because nobody has to worry about setting up environment variables and SqlAzure tables to run the ApiHub extensions tests for tabular connectors. The environment variable variable approach is obscure and counter-productive. Each team has to know about the other teams' specific details to be able to run tests. I would prefer to keep this until a better general solution exists. Removing stuff is usually easy.